Yoast/wordpress-seo
# Dynamically grant or revoke wpseo_manage_options cap for administrators

## Summary

This PR can be summarized in the following changelog entry:

* Ensure restricting SEO setting access to network admins only in multisite correctly revokes that capability from regular admins.

## Relevant technical choices:

* At the moment, this setting was only honored during plugin activation, because at that point capabilities and roles were set. The approach taken here removes the check for the `access` option from there and always grants `wpseo_manage_options` to administrators.
* The capability is then dynamically unset as necessary, using the `user_has_cap` filter. This ensures that a change of the setting is immediately reflected without requiring a plugin deactivation/reactivation.

## Test instructions

This PR can be tested by following these steps:

* Change the `access` option in the multisite SEO settings screen to the "Super Admins only" option.
* Ensure that, when that option is active, regular administrators who are not super admins/network administrators cannot access SEO settings.
* Also ensure that users with the more specific SEO manager role still _can_ access SEO settings regardless of the setting.

## Quality assurance


Fixes #5970
